prime-router/src/main/kotlin/Element.kt

```diff
@@ -843,20 +843,33 @@ data class Element(
         return retVal
     }
 
-    fun tokenizeMapperValue(elementName: String, index: Int = 0): ElementAndValue? {
+    fun tokenizeMapperValue(elementName: String, index: Int = 0): ElementAndValue {
         val tokenElement = Element(elementName)
-        return when (elementName) {
-            "\$index" -> {
-                ElementAndValue(tokenElement, index.toString())
+        var retVal = ElementAndValue(tokenElement, "")
+        when {
+            elementName == "\$index" -> {
+                retVal = ElementAndValue(tokenElement, index.toString())
             }
-            "\$currentDate" -> {
+            elementName == "\$currentDate" -> {
                 val currentDate = LocalDate.now().format(dateFormatter)
-                ElementAndValue(tokenElement, currentDate)
+                retVal = ElementAndValue(tokenElement, currentDate)
             }
-            else -> {
-                null
+            elementName.contains("\$dateFormat:") -> {
```

Comment:
@carlosfelix2 $mode and $string are tokens that will be used in my in-progress work on 2245, which I'm hoping to wrap up soon. I've removed them from this branch for now.

Reply:
Ah, ok.  But if you are removing it I think you need that else back in returning null, no?

Reply:
You will also need to update the PR description

Reply:
@carlosfelix2 actually $dateFormat is still relevant to this PR, and it uses extractStringValue(), so it would not be dead code. I'll have to confirm that that null case in the when is not needed. I'll remove refs to $mode and $string in the PR description and slap them back into my PR when it's ready, but I think $dateFormat is already explained in the PR, no? If you feel we need further explanation, I'll try and flesh that out a little more.

Reply:
@carlosfelix2 I confirmed that the null case is not needed. I believe I've addressed all your other comments, could you please re-review and (hopefully) approve?